lib/diff.ml

```diff
@@ -12,16 +12,25 @@ type value = {
   vdiff : (value_description, value_description atomic_modification) t;
 }
 
+type type_ = {
+  tname : string;
+  tdiff : (type_declaration, type_declaration atomic_modification) t;
+}
+
 type module_ = {
   mname : string;
   mdiff : (module_declaration, module_modification) t;
 }
 
 and module_modification = Unsupported | Supported of sig_item list
-and sig_item = Value of value | Module of module_
+and sig_item = Value of value | Module of module_ | Type of type_
```

Comment:
Note: an alternate design would have been to add the `Ident` only in the `Type` constructor payload, instead of on any entry of the `Sig_item_map`. That would have prevented adding the ident on values and module, where they are not used, and limited the conflicts on the rebase of other PRs!

But, your solution is still ok.

Reply:
Yes, that's a better design. but I think it should be more easily added to Typ constructor, and stored in ```Sig_item_map``` with Type_item item type only.